woocommerce/woocommerce
# Make OSA fields extendable, simplify naming

_This is an add-on to https://github.com/woocommerce/woocommerce/pull/39701_

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





- [Use `referrer` & `source_type` field names consistently](https://github.com/woocommerce/woocommerce/commit/1e293d9c4668b8f7f2081b65d44d753cc1440a1b) 
   Remove the need to translate it back and forth.
- [Make fields actually extendable](https://github.com/woocommerce/woocommerce/commit/8cfa6a2e6c452aa4a307a124977f7824365a5db5) 
   using `wc_order_attribution_tracking_fields`.
   Propagate the field configuration to the client side as well.
   
   Disambiguate `fields` in variables and functions.

### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Smoke test https://github.com/woocommerce/woocommerce/pull/39701 for regression
2. Add PHP snippet
```php

add_filter(
    'wc_order_attribution_tracking_fields',
    function( $fields ) {
        $fields['new_field'] = 'first_add.fd';
        return $fields;
    }
);
```
4. Visit classic checkout, 
   1. check that `<input type="hidden" name="wc_order_attribution_new_field" value="" />` is in the source code.
   2. Check its value is correctly set `document.querySelector('input[name="wc_order_attribution_new_field"]').value === sbjs.get.first_add.fd`
5. Visit Block checkout, and open dev tools
   1. click "Place order"
   2. Check `/checkout` ajax request payload contains the new field
      ![image](https://github.com/woocommerce/woocommerce/assets/17435/a9f3c0f5-4916-4389-8bba-4842b5bc7b77)
6. Check that order metadata contains the new property.
7. Review the code and check if all `field`-related names are understandable




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
